### ISSUE TITLE
PLUTO channel issue

### ISSUE THREAD

**Issue by Jag4343** [dtankdempse/free-iptv-channels#14]
I’m using the code for your Pluto and none of the channels are working. Getting a An error occurred: code 404 on every channel. I’m using it on Tivimate.

**Comment by Mikhel** [comment]
the .app.json file was deleted from the Matt Huisman's webpage

**Comment by dtankdempse** [comment]
Yes, I’m aware and currently working on a solution.

**Comment by s-digweed** [comment]
> Yes, I’m aware and currently working on a solution.

You are awesome. Bless you mate!

**Comment by Jag4343** [comment]
When you find a solution can you post the new code to use with tivimate? Samsung, Roku, & Plex is still working for me except for Pluto

**Comment by dtankdempse** [comment]
> When you find a solution can you post the new code to use with tivimate? Samsung, Roku, & Plex is still working for me except for Pluto

I have posted a temporary fix. Check the Read me section.

**Comment by Jag4343** [comment]
I went to the Read Me section. I don’t see the code to use. Can you send it to me. ThanksSent from my iPhoneOn Nov 5, 2024, at 3:51 PM, Reuben ***@***.***> wrote:﻿

When you find a solution can you post the new code to use with tivimate? Samsung, Roku, & Plex is still working for me except for Pluto

I have posted a temporary fix. Check the Read me section.

**Comment by dtankdempse** [comment]
Notice: PlutoTV is currently available only via the multiservice21 hosted script and provides a playlist limited to the US region. Since Matt has removed the Pluto service from his site, I'm working on a way to include additional regions. This setup is temporary until a more permanent solution is available. Additionally, Matt's EPG data is incompatible with this configuration, so please use the EPG found [here](https://raw.githubusercontent.com/dtankdempse/free-iptv-channels/refs/heads/main/pluto/us.xml).

**Comment by Jag4343** [comment]
Just checked again. Only the US channels seem to work. I will wait until you post another code. If you can post the code for US and all regions that would be great. Thanks againSent from my iPhoneOn Nov 5, 2024, at 4:17 PM, JEANS ***@***.***> wrote:﻿Seems like the Pluto channels are working againSent from my iPhoneOn Nov 5, 2024, at 4:16 PM, Reuben ***@***.***> wrote:﻿
Notice: PlutoTV is currently available only via the multiservice21 hosted script and provides a playlist limited to the US region. Since Matt has removed the Pluto service from his site, I'm working on a way to include additional regions. This setup is temporary until a more permanent solution is available. Additionally, Matt's EPG data is incompatible with this configuration, so please use the EPG found here.

**Comment by s-digweed** [comment]
> Notice: PlutoTV is currently available only via the multiservice21 hosted script and provides a playlist limited to the US region. Since Matt has removed the Pluto service from his site, I'm working on a way to include additional regions. This setup is temporary until a more permanent solution is available. Additionally, Matt's EPG data is incompatible with this configuration, so please use the EPG found [here](https://raw.githubusercontent.com/dtankdempse/free-iptv-channels/refs/heads/main/pluto/us.xml).

Ah ok. So the permanent solution also won't be compatible with Matt's epg?

**Comment by dtankdempse** [comment]
> > Notice: PlutoTV is currently available only via the multiservice21 hosted script and provides a playlist limited to the US region. Since Matt has removed the Pluto service from his site, I'm working on a way to include additional regions. This setup is temporary until a more permanent solution is available. Additionally, Matt's EPG data is incompatible with this configuration, so please use the EPG found [here](https://raw.githubusercontent.com/dtankdempse/free-iptv-channels/refs/heads/main/pluto/us.xml).
> 
> Ah ok. So the permanent solution also won't be compatible with Matt's epg?

Probably not. Since Matt has removed the JSON file and redirects for Pluto, it's uncertain how long he will continue supporting the Pluto EPGs

**Comment by s-digweed** [comment]
> > > Notice: PlutoTV is currently available only via the multiservice21 hosted script and provides a playlist limited to the US region. Since Matt has removed the Pluto service from his site, I'm working on a way to include additional regions. This setup is temporary until a more permanent solution is available. Additionally, Matt's EPG data is incompatible with this configuration, so please use the EPG found [here](https://raw.githubusercontent.com/dtankdempse/free-iptv-channels/refs/heads/main/pluto/us.xml).
> > 
> > 
> > Ah ok. So the permanent solution also won't be compatible with Matt's epg?
> 
> Probably not. Since Matt has removed the JSON file and redirects for Pluto, it's uncertain how long he will continue supporting the Pluto EPGs

This from matt btw
> intentional. another dmca. The author of the tinyurl script should be able to look at pluto kodi add-on latest changes and figure out what needs changing :)

**Comment by Mikhel** [comment]
> Notice: PlutoTV is currently available only via the multiservice21 hosted script and provides a playlist limited to the US region. Since Matt has removed the Pluto service from his site, I'm working on a way to include additional regions. This setup is temporary until a more permanent solution is available. Additionally, Matt's EPG data is incompatible with this configuration, so please use the EPG found [here](https://raw.githubusercontent.com/dtankdempse/free-iptv-channels/refs/heads/main/pluto/us.xml).



> > > Notice: PlutoTV is currently available only via the multiservice21 hosted script and provides a playlist limited to the US region. Since Matt has removed the Pluto service from his site, I'm working on a way to include additional regions. This setup is temporary until a more permanent solution is available. Additionally, Matt's EPG data is incompatible with this configuration, so please use the EPG found [here](https://raw.githubusercontent.com/dtankdempse/free-iptv-channels/refs/heads/main/pluto/us.xml).
> > 
> > 
> > Ah ok. So the permanent solution also won't be compatible with Matt's epg?
> 
> Probably not. Since Matt has removed the JSON file and redirects for Pluto, it's uncertain how long he will continue supporting the Pluto EPGs

There is something called "streamlink", it seems to be able to get EPG (XML) and M3U from different countries using IPs from different places but I might be wrong on how it is deployed and what it does (LOL):

https://github.com/streamlink/streamlink/blob/master/src/streamlink/plugins/pluto.py

**Comment by dtankdempse** [comment]
> > > > Notice: PlutoTV is currently available only via the multiservice21 hosted script and provides a playlist limited to the US region. Since Matt has removed the Pluto service from his site, I'm working on a way to include additional regions. This setup is temporary until a more permanent solution is available. Additionally, Matt's EPG data is incompatible with this configuration, so please use the EPG found [here](https://raw.githubusercontent.com/dtankdempse/free-iptv-channels/refs/heads/main/pluto/us.xml).
> > > 
> > > 
> > > Ah ok. So the permanent solution also won't be compatible with Matt's epg?
> > 
> > 
> > Probably not. Since Matt has removed the JSON file and redirects for Pluto, it's uncertain how long he will continue supporting the Pluto EPGs
> 
> This from matt btw
> 
> > intentional. another dmca. The author of the tinyurl script should be able to look at pluto kodi add-on latest changes and figure out what needs changing :)

This should work. I'm on it!

**Comment by Mikhel** [comment]
> This should work. I'm on it!

Thanks :)

**Comment by Jag4343** [comment]
Thanks for the update. Once everything is fixed could you let us know and what the new codes are that will work on Tivimate. Very Much appreciated!!! Sent from my iPhoneOn Nov 5, 2024, at 5:42 PM, Mikhel ***@***.***> wrote:﻿

This should work. I'm on it!

Thanks :)

**Comment by joaquinito2070** [comment]
You should use my friend repo https://github.com/[HelmerLuzo/PlutoTV_HL](https://github.com/HelmerLuzo/PlutoTV_HL) as that repo uses M3U and W3U. Also you will need to add https://github.com/[HelmerLuzo/RadTV_HL](https://github.com/HelmerLuzo/RadTV_HL), https://github.com/[HelmerLuzo/RakutenTV_HL](https://github.com/HelmerLuzo/RakutenTV_HL), https://github.com/[HelmerLuzo/FreeLiveSportsTV_HL](https://github.com/HelmerLuzo/FreeLiveSportsTV_HL), https://github.com/[HelmerLuzo/RuntimeTV_HL](https://github.com/HelmerLuzo/RuntimeTV_HL) and https://github.com/[HelmerLuzo/CanelaTV_HL](https://github.com/HelmerLuzo/CanelaTV_HL) M3U/W3U lists. SamsungTVPlus still same as HelmerLuzo SamsungTVPlus lists are Spain only, and the rest, in worldwide.

**Comment by Mikhel** [comment]
Yup although it would be a great addition to have the script to generate the channel lists appart from Helmer's lista which are quite great frankly.

**Comment by dtankdempse** [comment]
Ok, PlutoTV should be working again.

**Comment by Jag4343** [comment]
It’s working but is there a way to have just the countries and not the individual groups the way it is listed?Sent from my iPhoneOn Nov 5, 2024, at 8:05 PM, Reuben ***@***.***> wrote:﻿
Ok, PlutoTV should be working again.

**Comment by dtankdempse** [comment]
> It’s working but is there a way to have just the countries and not the individual groups the way it is listed?Sent from my iPhoneOn Nov 5, 2024, at 8:05 PM, Reuben ***@***.***> wrote:﻿ Ok, PlutoTV should be working again. —Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: ***@***.***>

Changed it to how it was.

**Comment by Jag4343** [comment]
Thanks. I see it now listed with countries. I do see that some of my channels are not there and some of the US EPG are in another language. If you can correct it I appreciate it. Sent from my iPhoneOn Nov 5, 2024, at 8:31 PM, Reuben ***@***.***> wrote:﻿

It’s working but is there a way to have just the countries and not the individual groups the way it is listed?Sent from my iPhoneOn Nov 5, 2024, at 8:05 PM, Reuben @.> wrote:﻿ Ok, PlutoTV should be working again. —Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: @.>

Changed it to how it was.

**Comment by dtankdempse** [comment]
> Thanks. I see it now listed with countries. I do see that some of my channels are not there and some of the US EPG are in another language. If you can correct it I appreciate it. Sent from my iPhoneOn Nov 5, 2024, at 8:31 PM, Reuben ***@***.***> wrote:﻿ It’s working but is there a way to have just the countries and not the individual groups the way it is listed?Sent from my iPhoneOn Nov 5, 2024, at 8:05 PM, Reuben @.> wrote:﻿ Ok, PlutoTV should be working again. —Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: @.> Changed it to how it was. —Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: ***@***.***>

The only EPG data in another language within the US region playlist that I can see is for Spanish channels. If any channels are missing from the playlist, they’re likely not included in Matt's JSON file. I also noticed some duplicate channels, so I'll work on fixing that.

**Comment by Jag4343** [comment]
This an example but the Pluto tv Christmas is not in English in the EPG. Look below at the picture. There are some of the other ones are like that in the US section. Sent from my iPhoneOn Nov 5, 2024, at 9:47 PM, Reuben ***@***.***> wrote:﻿

Thanks. I see it now listed with countries. I do see that some of my channels are not there and some of the US EPG are in another language. If you can correct it I appreciate it. Sent from my iPhoneOn Nov 5, 2024, at 8:31 PM, Reuben @.> wrote:﻿ It’s working but is there a way to have just the countries and not the individual groups the way it is listed?Sent from my iPhoneOn Nov 5, 2024, at 8:05 PM, Reuben @.> wrote:﻿ Ok, PlutoTV should be working again. —Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: @.> Changed it to how it was. —Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: @.>

The only EPG data in another language within the US region playlist that I can see is for Spanish channels. If any channels are missing from the playlist, they’re likely not included in Matt's JSON file. I also noticed some duplicate channels, so I'll work on fixing that.

**Comment by Jag4343** [comment]
Here are a few more examples below  Some of the EPG for the US, CA and the  United Kingdom are in another language. Before the issue yesterday the EPG was in English. Take look through all 3 of those countries and you will see. Thanks for all the work you’re doing. Sent from my iPhoneOn Nov 5, 2024, at 9:47 PM, Reuben ***@***.***> wrote:﻿

Thanks. I see it now listed with countries. I do see that some of my channels are not there and some of the US EPG are in another language. If you can correct it I appreciate it. Sent from my iPhoneOn Nov 5, 2024, at 8:31 PM, Reuben @.> wrote:﻿ It’s working but is there a way to have just the countries and not the individual groups the way it is listed?Sent from my iPhoneOn Nov 5, 2024, at 8:05 PM, Reuben @.> wrote:﻿ Ok, PlutoTV should be working again. —Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: @.> Changed it to how it was. —Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: @.>

The only EPG data in another language within the US region playlist that I can see is for Spanish channels. If any channels are missing from the playlist, they’re likely not included in Matt's JSON file. I also noticed some duplicate channels, so I'll work on fixing that.

**Comment by Mikhel** [comment]
> Here are a few more examples below  Some of the EPG for the US, CA and the  United Kingdom are in another language. Before the issue yesterday the EPG was in English. Take look through all 3 of those countries and you will see. Thanks for all the work you’re doing. 

I just checked and I don't see what you are seeing, check you have the correct URL for the EPG for those lists.

For Pluto USA what channels do you see with an EPG in a different language?

**Comment by Jag4343** [comment]
I am using the same url that I have before that is listed on your page. When the issue happened last night and then you said it was fixed today I continue to update my playlist and EPG. Some of the 90210, Pluto Tv Christmas, etc are not in Englishin the EPG. Its not all of them but below take a look at my pics and here are both of the URL’s that I’m using and some examples of the EPG and descriptions that are not in English. Sent from my iPhoneOn Nov 5, 2024, at 10:18 PM, Mikhel ***@***.***> wrote:﻿

Here are a few more examples below  Some of the EPG for the US, CA and the  United Kingdom are in another language. Before the issue yesterday the EPG was in English. Take look through all 3 of those countries and you will see. Thanks for all the work you’re doing. 

I just checked and I don't see what you are seeing, check you have the correct URL for the EPG for those lists.
For Pluto USA what channels do you see with an EPG in a different language?

**Comment by dtankdempse** [comment]
You're right; there was an issue I missed when using the 'all' region. The channel IDs needed to be unique to prevent conflicts. Try it now and let me know how it goes. Thanks!

**Comment by Jag4343** [comment]
I also noticed for example that dog the bounty hunter says in the EPG that each channel in each country has the same episode on but when you go to each channel there is a different episode on. Before the issue yesterday each country had a different episode and the EPG was correct. The EPG is not working properly for each of those dog the bounty hunter channels. You will see below that each channel has the same EPG episode on each channel. Thats not correct. You will see below in my picSent from my iPhoneOn Nov 5, 2024, at 10:33 PM, JEANS ***@***.***> wrote:﻿I am using the same url that I have before that is listed on your page. When the issue happened last night and then you said it was fixed today I continue to update my playlist and EPG. Some of the 90210, Pluto Tv Christmas, etc are not in Englishin the EPG. Its not all of them but below take a look at my pics and here are both of the URL’s that I’m using and some examples of the EPG and descriptions that are not in English. <IMG_7454.jpeg><IMG_7455.jpeg><IMG_7450.jpeg><IMG_7451.jpeg><IMG_7452.jpeg><IMG_7453.jpeg>Sent from my iPhoneOn Nov 5, 2024, at 10:18 PM, Mikhel ***@***.***> wrote:﻿

Here are a few more examples below  Some of the EPG for the US, CA and the  United Kingdom are in another language. Before the issue yesterday the EPG was in English. Take look through all 3 of those countries and you will see. Thanks for all the work you’re doing. 

I just checked and I don't see what you are seeing, check you have the correct URL for the EPG for those lists.
For Pluto USA what channels do you see with an EPG in a different language?

**Comment by Mikhel** [comment]
> You're right; there was an issue I missed when using the 'all' region. The channel IDs needed to be unique to prevent conflicts. Try it now and let me know how it goes. Thanks!

Ah, I manually assigned the EPG to each list, didn't check if it was asking for all.xml.gz, for USA I used us.xml.gz and so on. That's why I am not seeing any discrepancies.

**Comment by dtankdempse** [comment]
> > You're right; there was an issue I missed when using the 'all' region. The channel IDs needed to be unique to prevent conflicts. Try it now and let me know how it goes. Thanks!
> 
> Ah, I manually assigned the EPG to each list, didn't check if it was asking for all.xml.gz, for USA I used us.xml.gz and so on. That's why I am not seeing any discrepancies.

Right, the issue @Jag4343 mentioned only occurs when the region is set to 'all.' I actually solved this in the previous script, but it completely slipped my mind when coding this new function.

**Comment by Jag4343** [comment]
I am not very tech savvy and don’t understand to much of all of this but I have had a favorite list on Tivimate and when the issue happened I noticed that a lot of channels EPG wasn’t in the correct language and the EPG said that each channel in each country had the same episode so I knew something wasn’t correct. It seems like it is corrected but I will take a look closer. The only think I see that is not correct is that maybe about 10 channels that I had in m favorites disappeared so maybe like you said that those channels are not in your list. Like I said, I don’t know much about the programming end but I just know what I saw that wasn’t correct. I will take a look closer but it seems like it’s fixed except for the missing channels. Thanks for all of your work and time. We use Pluto and I appreciate it. ThanksSent from my iPhoneOn Nov 5, 2024, at 10:44 PM, Mikhel ***@***.***> wrote:﻿

You're right; there was an issue I missed when using the 'all' region. The channel IDs needed to be unique to prevent conflicts. Try it now and let me know how it goes. Thanks!

Ah, I manually assigned the EPG to each list, didn't check if it was asking for all.xml.gz, for USA I used us.xml.gz and so on. That's why I am not seeing any discrepancies.

**Comment by dtankdempse** [comment]
> I am not very tech savvy and don’t understand to much of all of this but I have had a favorite list on Tivimate and when the issue happened I noticed that a lot of channels EPG wasn’t in the correct language and the EPG said that each channel in each country had the same episode so I knew something wasn’t correct. It seems like it is corrected but I will take a look closer. The only think I see that is not correct is that maybe about 10 channels that I had in m favorites disappeared so maybe like you said that those channels are not in your list. Like I said, I don’t know much about the programming end but I just know what I saw that wasn’t correct. I will take a look closer but it seems like it’s fixed except for the missing channels. Thanks for all of your work and time. We use Pluto and I appreciate it. ThanksSent from my iPhoneOn Nov 5, 2024, at 10:44 PM, Mikhel ***@***.***> wrote:﻿ You're right; there was an issue I missed when using the 'all' region. The channel IDs needed to be unique to prevent conflicts. Try it now and let me know how it goes. Thanks! Ah, I manually assigned the EPG to each list, didn't check if it was asking for all.xml.gz, for USA I used us.xml.gz and so on. That's why I am not seeing any discrepancies. —Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: ***@***.***>

Send me a list of the missing channels and I will check.

**Comment by Jag4343** [comment]
I went through all of my previously saved favorites in my list and I had over 100 of them. When the issue occurred last night and then you fixed it and I updated my playlist and the EPG I was missing 3 channels that are not there, maybe about 15 channels that were previously saved in favorites disappeared but were still available so I re saved them as a favorite. The 3 channels which were there before but are totally missing are:1. Hallmark Movies & More (US)2. Movie Favorites by Lifetime (US)3. Pluto TV Stand Up (GB) Not a big deal if you can’t find them. I did also notice that previously the country was listed as GREAT BRITAIN and when you fixed it it says UNITED KINGDOM. I know it’s the same but some of my channels did delete from my favorite list. Some were US and Canada channels so I’m not sure if this was normal for them to go away. Thanks!!!Sent from my iPhoneOn Nov 5, 2024, at 10:59 PM, Reuben ***@***.***> wrote:﻿

I am not very tech savvy and don’t understand to much of all of this but I have had a favorite list on Tivimate and when the issue happened I noticed that a lot of channels EPG wasn’t in the correct language and the EPG said that each channel in each country had the same episode so I knew something wasn’t correct. It seems like it is corrected but I will take a look closer. The only think I see that is not correct is that maybe about 10 channels that I had in m favorites disappeared so maybe like you said that those channels are not in your list. Like I said, I don’t know much about the programming end but I just know what I saw that wasn’t correct. I will take a look closer but it seems like it’s fixed except for the missing channels. Thanks for all of your work and time. We use Pluto and I appreciate it. ThanksSent from my iPhoneOn Nov 5, 2024, at 10:44 PM, Mikhel @.> wrote:﻿ You're right; there was an issue I missed when using the 'all' region. The channel IDs needed to be unique to prevent conflicts. Try it now and let me know how it goes. Thanks! Ah, I manually assigned the EPG to each list, didn't check if it was asking for all.xml.gz, for USA I used us.xml.gz and so on. That's why I am not seeing any discrepancies. —Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: @.>

Send me a list of the missing channels and I will check.

**Comment by s-digweed** [comment]
Pluto is fixed, Samsung works fine too. Big kudos Reuben @dtankdempse

**Comment by Mikhel** [comment]
> Is the selfhosted gscript supposed to be working? Because it fails with Pluto atm.

Everything is working fine with me, I just modified my self hosted script with the one by DTankDempse. His is also working fine:

https://tinyurl.com/multiservice21?region=us&service=PlutoTV

Saludos.

**Comment by dtankdempse** [comment]
> > > Is the selfhosted gscript supposed to be working? Because it fails with Pluto atm.
> > 
> > 
> > Everything is working fine with me, I just modified my self hosted script with the one by DTankDempse. His is also working fine:
> > https://tinyurl.com/multiservice21?region=us&service=PlutoTV
> > Saludos.
> 
> Tinyurl works fine, but https://github.com/dtankdempse/free-iptv-channels/blob/main/code.gs does not [only with Pluto]

The updated code.gs should be working. Have you redeployed the Google App Script and are you using the new deployment URL?
